File: Source/UserManagement/Web.Angular/src/app/domain/data-collector/ChangeBaseInformation.ts
import { Command } from '../../services/Command';
import { Sex } from '../sex';


export class ChangeBaseInformation extends Command {
    dataCollectorId: string;
    fullName: string;
    displayName: string;
    yearOfBirth: number;
    sex: number;

    region: string;
    district: string;

    constructor() {
        super();
        this.type = 'CBS#UserManagement.DataCollector.Changing-ChangeBaseInformation+Command@1';
    }
}

File: Source/VolunteerReporting/Domain/DataCollectors/Changing/ChangeBaseInformation.cs
/*---------------------------------------------------------------------------------------------
 *  Copyright (c) The International Federation of Red Cross and Red Crescent Societies. All rights reserved.
 *  Licensed under the MIT License. See LICENSE in the project root for license information.
 *--------------------------------------------------------------------------------------------*/

using Concepts.DataCollector;
using Dolittle.Commands;

namespace Domain.DataCollectors.Changing
{
    public class ChangeBaseInformation : ICommand
    {
        public DataCollectorId DataCollectorId { get; set; }

        public string FullName { get; set; }
        public string DisplayName { get; set; }
        //TODO: Add later on. public string Email { get; set; }
        public int YearOfBirth { get; set; }
        public Sex Sex { get; set; }

        public string Region { get; set; }
        public string District { get; set; }
    }
}
